### PR TITLE
CL: Support waiting for arbitrary objects after creation

### PR DESCRIPTION
**Pull request by sjug**
The new concurrent created templates should also wait for object to be in good status.

### CODE REVIEW

File: test/extended/cluster/cl.go [hunk 1]
```diff
@@ -222,75 +223,8 @@ var _ = g.Describe("[Feature:Performance][Serial][Slow] Load cluster", func() {
...
-			}
-			if len(dcList.Items) > 0 {
-				// Get first deployment config name
-				deploymentName := dcList.Items[0].Name
-				e2e.Logf("Waiting for deployment: %q", deploymentName)
-				err = exutil.WaitForDeploymentConfig(oc.AdminKubeClient(), oc.AsAdmin().AppsClient().AppsV1(), ns, deploymentName, 1, true, oc)
-				o.Expect(err).NotTo(o.HaveOccurred())
-				e2e.Logf("Deployment %q completed", deploymentName)
-			}
+		if err := postWait(oc, namespaces); err != nil {
```
Comment (jmencak) [inline test/extended/cluster/cl.go]: Can we rename postWait() to give it a more descriptive meaning?  A few comments at the top of the function would also help.

File: test/extended/cluster/clt.go [hunk 2]
```diff
@@ -45,27 +47,35 @@ var _ = g.Describe("[Feature:Performance][Serial][Slow] Load cluster", func() {
...
-			go clWorker(work, &wg, oc)
+			go clWorker(work, ns, &wg, oc)
 		}
 
 		for _, p := range project {
 			for j := 0; j < p.Number; j++ {
 				wg.Add(1)
 				unit := ProjectMeta{j, p, viperConfig}
 				work <- unit
+				namespace := <-ns
```
Comment (jmencak) [inline test/extended/cluster/clt.go]: Clever use of channels!

File: test/extended/cluster/cl.go [hunk 3]
```diff
@@ -309,6 +243,83 @@ var _ = g.Describe("[Feature:Performance][Serial][Slow] Load cluster", func() {
 	})
 })
 
+func postWait(oc *util.CLI, namespaces []string) error {
+	// Wait for builds and deployments to complete
+	for _, ns := range namespaces {
+		rcList, err := oc.AdminKubeClient().CoreV1().ReplicationControllers(ns).List(metav1.ListOptions{})
+		if err != nil {
+			return err
```
Comment (jmencak) [inline test/extended/cluster/cl.go]: I wonder why the more specific error messages were dropped after the code was lifted into this function.  Are the generic err(s) that you return enough to tell the user what happened?

File: test/extended/cluster/cl.go (hunk 1 above)
Comment (sjug) [reply test/extended/cluster/cl.go]: How's `postCreateWait()`

File: test/extended/cluster/cl.go (hunk 3 above)
Comment (sjug) [reply test/extended/cluster/cl.go]: Wanted to pass the errors back to be handled in the main function, passing these new errors instead. What do you think?

**Pull request merged**